src/extensions/default/CodeFolding/Prefs.js

```diff
@@ -85,8 +93,9 @@ define(function (require, exports, module) {
       * @param {Object} folds the fold ranges to save for the current document
       */
     function setFolds(path, folds) {
-        store[path] = simplify(folds);
-        prefs.set(foldsKey, store);
+        var allFolds = PreferencesManager.getViewState(foldsKey);
```

Comment:
One other thought... rather than dumping folding state for all files together in one map, it might make sense to bucket it by project, like we do for other similar prefs like the scroll position & selection state of editors.

MainViewManager has an example of how to do that:
```
            context = { location : { scope: "user",
                                     layer: "project" } },
            state = PreferencesManager.getViewState(PREFS_NAME, context);
```

and

```
            context         = { location : { scope: "user",
                                         layer: "project",
                                         layerID: projectRoot.fullPath } },
        ...
        PreferencesManager.setViewState(PREFS_NAME, state, context);
```